cmd/geth/flags.go

```diff
@@ -152,8 +152,37 @@ var (
 	}
 	LogDirFlag = DirectoryFlag{
 		Name:  "log-dir,logdir",
-		Usage: "Directory in which to write log files.",
-		Value: DirectoryString{filepath.Join(common.DefaultDataDir(), "logs")},
+		Usage: "Directory in which to write log files",
+		Value: DirectoryString{filepath.Join(common.DefaultDataDir(), "<chain>", "logs")},
+	}
+	LogMaxSizeFlag = cli.IntFlag{
+		Name:  "log-max-size,log-maxsize",
+		Usage: "Maximum size of a single log file (in bytes)",
+		Value: 0,
+	}
+	LogMinSizeFlag = cli.IntFlag{
+		Name:  "log-min-size,log-minsize",
+		Usage: "Minimum size of a log file, to be considered for log-rotation (in bytes)",
+		Value: 0,
+	}
+	LogMaxTotalSizeFlag = cli.IntFlag{
+		Name:  "log-max-total-size,log-maxtotalsize",
+		Usage: "Maximum total size of all (current and archived) log files (in bytes)",
+		Value: 0,
+	}
+	LogIntervalFlag = cli.StringFlag{
+		Name:  "log-interval",
+		Usage: "Log rotation interval, one of values: never, hourly, daily, weekly, monthly",
+		Value: "never",
+	}
+	LogMaxAgeFlag = cli.StringFlag{
+		Name:  "log-max-age,log-maxage",
+		Usage: "Maximum age of the oldest log file, valid time units: 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'",
+		Value: "0",
+	}
+	LogCompressFlag = cli.BoolFlag{
```

Comment:
Should we make this a `BoolTFlag` to enable by default?

I guess this would entail establishing defaults for max/min size and age as well, but if we're going to be generating the logs anyways it might be reasonable.

Reply:
I didn't want to change the default behavior of keeping logs forever by default.
Enabling compression by default seems very reasonable, min/max size also, but I'm not sure about the max age and total size.
My new proposition for default values:
 - compression: true
 - rotation-interval: hourly,
 - min-size: 1MB or less?
 - max-size: I think 10-20 MB should be OK for hourly rotaiton interval (I was running with 50MB limit for few hours, and only time-based rotation happened)
 - max-age, total-max-size - 0 (keep logs forever)

This gives usable logs - easy to seek by date/time, fine-grained if we wan't to remove old ones, compressed files are quite small so they are easy to search even with editor (not only `grep`). 

I think we should run `geth` version with output/log split, with default level/verbosity/etc, for some time, and find sane defaults for log rotation empirically.

Reply:
Those defaults seem very reasonable to me. And agree that we should get an empirical sense.